tempesta_fw/http.c

```diff
@@ -821,9 +821,14 @@ __http_req_delist(TfwSrvConn *srv_conn, TfwHttpReq *req)
 }
 
 static inline void
-tfw_http_req_delist(TfwSrvConn *srv_conn, TfwHttpReq *req)
+tfw_http_req_delist(TfwHttpMsg *hmresp)
 {
+	TfwHttpReq *req = hmresp->req;
+	TfwSrvConn *srv_conn = (TfwSrvConn *)hmresp->conn;
+
 	spin_lock(&srv_conn->fwd_qlock);
+	if ((TfwMsg *)req == srv_conn->msg_sent)
+		srv_conn->msg_sent = NULL;
```

Comment:
Incidentally, this is exactly what's done in `tfw_http_popreq()`. If this change is here to stay, then this sequence in that function can be replaces with a call to `tfw_http_req_delist()`.

I think the question I have is whether this check should be here or in `_tfw_http_req_delist()`. If it's the latter, then are there cases where this check is unnecessary?